recipes/mp-units/2.3.0/conanfile.py

```diff
@@ -0,0 +1,325 @@
+        if self.options.import_std:
+            if self.options.contracts != "none":
+                raise ConanInvalidConfiguration(
+                    "'contracts' should be set to 'none' to use `import std;`"
+                )
+            if not self.options.get_safe("std_format", default=True):
+                raise ConanInvalidConfiguration(
+                    "'std_format' should be enabled to use `import std;`"
+                )
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        opt = self.options
+        tc = CMakeToolchain(self)
+        if opt.cxx_modules:
+            tc.cache_variables["CMAKE_CXX_SCAN_FOR_MODULES"] = True
+        if opt.import_std:
+            tc.cache_variables["CMAKE_CXX_MODULE_STD"] = True
+            # TODO current experimental support according to `Help/dev/experimental.rst`
+            tc.cache_variables["CMAKE_EXPERIMENTAL_CXX_IMPORT_STD"] = (
+                "0e5b6991-d74f-4b3d-a41c-cf096e0b2508"
+            )
+
+        # TODO remove the below when Conan will learn to handle C++ modules
+        if opt.freestanding:
+            tc.cache_variables["MP_UNITS_API_FREESTANDING"] = True
+        else:
+            tc.cache_variables["MP_UNITS_API_STD_FORMAT"] = opt.std_format
+        tc.cache_variables["MP_UNITS_API_CONTRACTS"] = str(opt.contracts).upper()
+
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(build_script_folder="src")
+        if self.options.cxx_modules:
+            cmake.build()
+
+    def package_id(self):
+        if self.package_type == "header-library":
+            self.info.clear()
+
+    def package(self):
+        copy(
+            self,
+            "LICENSE.md",
+            self.source_folder,
+            os.path.join(self.package_folder, "licenses"),
+        )
+        cmake = CMake(self)
+        cmake.install()
+        # TODO remove the below when Conan will learn to handle C++ modules
+        if not self.options.cxx_modules:
+            # We have to preserve those files for C++ modules build as Conan
+            # can't generate such CMake targets for now
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        compiler = self.settings.compiler
+        # TODO remove the branch when Conan will learn to handle C++ modules
+        if self.options.cxx_modules:
+            # CMakeDeps does not generate C++ modules definitions for now
+            # Skip the Conan-generated files and use the mp-unitsConfig.cmake bundled with mp-units
+            self.cpp_info.set_property("cmake_find_mode", "none")
+            self.cpp_info.builddirs = ["."]
+        else:
```

Comment:
please note that an `else` may not be required here - at least not ideally. That is, the above cases will affect only the CMakeToolchain/CMakeDeps combo, but _other_ generators may still benefit from having the information, otherwise they won't have any information at all. However, given that this is modules, realistically it only works with CMake, so it doesn't have practical implications

Reply:
C++ modules are off for older compilers and can be disabled by the user as well with a dedicated option. In such a case, CMake is not a constraint for our users, and they can use any build systems/generator they want. This is when `else` branch may com handy ;-)